Microsoft.Alm.Authentication/Microsoft.Alm.Authentication.csproj

```diff
@@ -99,11 +110,26 @@
     <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
+  <Import Project="$(PackagesDirectory)\MSBuildTasks.*\tools\MSBuild.Community.Tasks.targets" />
+
+  <!-- Produce a nuget package of this assembly -->
+  <Target Name="AfterBuild" Condition=" '$(Configuration)' == 'Package'">
+    <!-- Only download a new copy of nuget.exe if we don't have a copy available -->
+    <PropertyGroup Condition="!Exists('nuget.exe')">
+      <HaveLocalNuget>true</HaveLocalNuget>
+    </PropertyGroup>
+    <PropertyGroup Condition="!$(HaveLocalNuget)">
+      <NugetPath>$(IntermediateOutputPath)nuget.exe</NugetPath>
+    </PropertyGroup>
+    <PropertyGroup Condition="$(HaveLocalNuget)">
+      <NugetPath>nuget.exe</NugetPath>
+    </PropertyGroup>
+
+    <WebDownload
+      Condition="!$(HaveLocalNuget) And !Exists($(NugetPath))"
+      Filename="$(NugetPath)"
+      FileUri="https://dist.nuget.org/win-x86-commandline/latest/nuget.exe" />
+    <Exec
+      Command="$(NugetPath) pack $(AssemblyName).csproj -Symbols -IncludeReferencedProjects -Prop Configuration=Release -OutputDirectory $(OutputPath)" />
```

Comment:
No idea. MSBuild is black magic to me. I just sacrifice the herbs and animal fat as instructed, and hope for best results.